packages/@atjson/document/src/annotation.ts

```diff
@@ -49,6 +49,30 @@ export default abstract class Annotation<Attributes = {}> {
            this.end === annotation.end;
   }
 
+  compareAttributes(lhsAnnotationAttributes: any, rhsAnnotationAttributes: any) {
+    return Object.keys(lhsAnnotationAttributes).every(function (key) {
+      if (lhsAnnotationAttributes[key] !== rhsAnnotationAttributes[key]) {
+        if (lhsAnnotationAttributes[key] instanceof Document && rhsAnnotationAttributes[key] instanceof Document) {
+          return lhsAnnotationAttributes[key].equal(rhsAnnotationAttributes[key])
+        }
+        if (lhsAnnotationAttributes[key] typeof Object && rhsAnnotationAttributes[key] typeof Object) {
```

Comment:
I'm not sure if this will work, since `typeof lhsAnnotationAttributes[key] === 'object'` is the typical pattern for this.